### PR TITLE
User Profile: Initial beta overview documentations

### PR DESCRIPTION
**Pull request by ywangd**
Add initial feature overview documentation pages for User Profile and
Security Domain.

Docs preview links:
* [User profile](https://elasticsearch_83766.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/user-profile.html)
* [Security domain](https://elasticsearch_83766.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-domain.html)

Relates to #83495

### CODE REVIEW

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 1]
```diff
@@ -0,0 +1,86 @@
+[role="xpack"]
+[[user-profile]]
+=== User Profile
+
+beta::[]
+
+User Profile provides a way to uniquely and stably refer to a user.
+Unique means each user has an identifier that is guaranteed to be unique within
+a deployment. Stable means the identifier remains generally unchanged in the
+lifetime of a deployment.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/user-profile.asciidoc:10]: I think this starts on the wrong footing. Are users not unique enough that we need profiles for it?

**Review by albertzaharovits — COMMENTED** [review]
I believe this will need a couple of rounds, and Adam's input is crucial.

This feature is particularly hard to explain to users, because it presumes familiarity with authentication realms, the general problem with listing users for realms, and ES ownership.

And the above says nothing about Kibana's intent with this feature for user personalization which influenced the design.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+request, <<security-api-create-api-key,creating an API key>>, or activating a 
+<<user-profile,user profile>>.
+
+When a user attempts to access a resource, {es} compares a user's `username`
+and the realm they're authenticating from to determine who they are. In some
+cases, the same user can authenticate with different realms. For example,
+an <<ldap-realm,LDAP realm>> and a <<saml-realm,SAML realm>> can be backed by
+the same directory service. Similarly, a SAML realm might
+<<configuring-authorization-delegation,delegate authorization>> to a
+<<native-realm,native realm>>.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:21]: @ywangd Is this supposed to refer to the fact that users authn to a SAML realm which delegates authorization to an LDAP realm, are viewed as different users if they would directly authn to the LDAP realm, because I don't believe this is the case.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+If {es} detects that a `username` from two different realms is attempting to
+access a resource, {es} assumes that these users are unique and doesn't allow
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:24]:
```suggestion
access a resource, {es} assumes that these users are distinct and doesn't allow
````

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+resources to be shared between them. With security domains, you can associate a 
+single <<user-profile,user profile>> with a `username` to enable resource sharing.
+This means that user preferences, notifications, and other user data can be
+shared across domains. The same user can view results from an asynchronous
+search request or a scrolling search, regardless of which domain they're
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:29]:
```suggestion
search request or a scrolling search, regardless of which realm under the same domain they're 
```

**Comment by ywangd** [comment]
> I believe this will need a couple of rounds, and Adam's input is crucial.
> 
@albertzaharovits I agree. It should take as long as it needs. Writing overview docs for this feature is a real challenge for me. I tried and scratched a couple times and it still ended up to be very inside-out. 

@lockewritesdocs Please feel free to make any changes that you see fit. Hopefully it still provides useful (technical) information for you to tell the story in a better way.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc (hunk 1 above)
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc]: I agree it's a bad openning statement because I didn't try to imply that users are not unique enough. In fact, it is quite the opposite, users (username + realm) are too unique. But we probably don't want these details in the beginning either. So I think the missing keyword here is "persistent" because users can be empheral but profiles are always persisted.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+This may seem to be trivial but it underpins many other useful features.
+For example, preferences can be customized and persisted separately for each user,
+messages and notifications can be routed to the right user using its unique identifier.
+
+The {stack} has the notion of "user" which is required to authenticate each request.
+Internally, the user is represented as a `User` object that encapsulates its identity
+and privileges. However, the `User` object itself is not sufficient to support referring
+to a user because:
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc]: No, we don't. We should strive to convey the same/similar message without having to resort to the `User` object.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+and the realm they're authenticating from to determine who they are. In some
+cases, the same user can authenticate with different realms. For example,
+an <<ldap-realm,LDAP realm>> and a <<saml-realm,SAML realm>> can be backed by
+the same directory service. Similarly, a SAML realm might
+<<configuring-authorization-delegation,delegate authorization>> to a
+<<native-realm,native realm>>.
+
+If {es} detects that a `username` from two different realms is attempting to
+access a resource, {es} assumes that these users are unique and doesn't allow
+resources to be shared between them. With security domains, you can associate a
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc:25]: At high level, associating a single user profile to multiple logins of the same user is also resource sharing. So I don't consider it being a separate functions, but rather one concrete application of the function, other applications are async search result, API keys etc.

**Comment by ywangd** [comment]
> And the above says nothing about Kibana's intent with this feature for user personalization which influenced the design.

This is something that is always not clear to me: How much should we bring Kibana and even solution in context when documenting a feature? I feel that if we are just documenting for Elasticsearch, it should be kept as neutral as possible. So I was actually actively trying to avoid bringing in anything about Kibana etc. But this could be a wrong position. I am really not sure.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc (hunk 5 above)
[the remark this answers is not in the document]
Comment (lockewritesdocs) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc]: I agree! I removed mention of the `User` object in 485baa5e7d02276f6003bd9d72d7098a7b1b28d6.

**Comment by lockewritesdocs** [comment]
@ywangd and @albertzaharovits -- I tried to refocus the User profiles page in 5ca168de03af2e86a2d4c1b15303ef55885f556b to explain how user profiles differ from user names and what their advantages are. 

>How much should we bring Kibana and even solution in context when documenting a feature? I feel that if we are just >documenting for Elasticsearch, it should be kept as neutral as possible. 

Although this is an Elasticsearch feature, we should think of it in context of the Stack, such as where it's used. From what I understand in talking to @albertzaharovits, persistence across sessions in Kibana is an important feature of user profiles, so we should call it out.

**Comment by ywangd** [comment]
@elasticmachine update branch

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+separately for each user, which is key to fine-grained level of customisation.
+This unique ID also offers a way to uniquely refer to a user so that
+{kib} can route messages and notifications to the appropriate user,
+regardless of whether they're logged in.
+
+===== Why a new unique ID instead of just the `username`?
+
+The `username` is _not_ unique outside a single realm. Using just the `username`
+is not possible to differentiate two users with the same `username` but from
+different realms.
```
Comment (ywangd) [inline x-pack/docs/en/security/authentication/user-profile.asciidoc:43]: A separate unique ID also makes it somewhat easier to support `username` changes. But I leave it out since this possibility hasn't really been explored.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+functionality.
+
+==== Limitations
+
+* Creating a new user profile requires a user's authentication details
+(`username` and `password` or its
+<<token-authentication-services,OAuth2 access token>>).
+This means that a user must authenticate at least one time to create a
+profile document. If a user hasn't authenticated into the system,
+then it's not possible to refer to that user with user profile.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/user-profile.asciidoc:79]:
```suggestion
profile document. Users who have never authenticated to {kib}
(or another profile aware application) will not have a user profile
and will not be returned by the 
<<security-api-search-user-profile,search user profile API>> 
```

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+the lifetime of a deployment and can be used to refer to the associated
+user.
+Being able to uniquely refer to users is a critical function that underpins
+important features like personalisation and collaboration in {kib}.
+
+==== How user profile works
+
+In the {stack}, a user is required to authenticate each request. Each user is
+represented as an object that encapsulates its identity and privileges.
+This object is ephemeral especially for users authenticating from a realm
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc:20]: I meant to use `object` for its vanilla meaning, that is, a thing. But I see how it can be interpreted as programming term in a doc like this.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+When a user logs in to {kib}, {kib} creates a new profile or associates an
+existing one for the user.
+By using the unique ID of the user profile, {kib} can store user preferences
+separately for each user, which is key to fine-grained level of customisation.
+This unique ID also offers a way to uniquely refer to a user so that
+{kib} can route messages and notifications to the appropriate user,
+regardless of whether they're logged in.
+
+===== Why a new unique ID instead of just the `username`?
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc:40]: Good point. Thanks

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+Persistent means it exists without the need for the user to be online.
+Stable means it has an unique identifier that does not change through
+the lifetime of a deployment and can be used to refer to the associated
+user.
+Being able to uniquely refer to users is a critical function that underpins
+important features like personalisation and collaboration in {kib}.
+
+==== How user profile works
+
+In the {stack}, a user is required to authenticate each request. Each user is
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc:18]: Thanks for rewriting this section. I adopted with a minor edit: Instead of saying a profile is associated with "one or more users". I changed to it to say "is associated to a user". I think it is confusing at this point to hint that profile support the same user from different realms. We do talk about it in the following section after taking about the basic. So I'll keep this section a bit simpler.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+==== Create and manage user profiles
+
+To create a new user profile or update an existing one, use the
+<<security-api-activate-user-profile,activate user profile API>>. When you
+submit a request, {es} attempts to locate an existing profile document for the
+specified user. If one doesn't exist, {es} creates a new profile document.
+
+In either case, the profile document captures the user's `full_name`, `email`,
+`roles`, and `realms`, and also includes the profile unique ID, timestamp for
+the operation, and version control numbers. You can retrieve a user profile with
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc:62]: I replaced it with `etc.`.

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+Because the {stack} has support for externally managed users (such as users who authenticate
+via SAML, or users stored in an LDAP directory), we make a distinction between the _user_ and their
+_profile_.
+
+The term "users" refer to the entities that authenticate requests to the {stack}.
+Each user has a username and a set of privileges (represented by <<roles,roles>>) that
+determine which types of requests may be performed by that user.
+This _user_ may be ephemeral - it could exist only for the duration of a request to
+an {es} API or for the lifetime of a session in {kib}. It cannot be retrieved once
+the session is complete, and does not support storage of preferences.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/user-profile.asciidoc:32]:
```suggestion
an {es} API or for the lifetime of a session in {kib}. These ephemeral users 
cannot be retrieved once the session is complete, and do not support storage
of preferences.
```

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+A "user profile" is the long term record associated with a _user_.
+It has a unique identifier, it is searchable and it can be used to store personalisation
+data and preferences for the user. It is a representation of the _user_ that persists
+and even when the _user_ itself has no active requests or sessions in the {stack}.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/user-profile.asciidoc:37]:
```suggestion
even when the _user_ itself has no active requests or sessions in the {stack}.
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 15]
```diff
@@ -0,0 +1,97 @@
+[role="xpack"]
+[[security-domain]]
+=== Security domains
+
+beta::[]
+
+Security domains are a method of grouping <<realms,realms>> to enable resource
+sharing across different realms for the same user.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:8]: This wording "different realms for the same user" is a little awkward, but it's hard to come up with something better.

I think perhaps the problem is trying to describe it as "resource sharing" which is a little too much _implementation_. If we try to approach it from a conceptual point of view we can word things differently.
 
```suggestion
Security domains are a method of grouping <<realms,realms>> so that the {stack}
can recognize when a single user authenticates to two different realms.
```
I still don't love it, but I think it's easier to introduce the concept that way first.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+
+==== How security domain works
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:10]:
```suggestion
==== How security domains work
```

Alternatively "How __*a*__ security domain works", but that doesn't read as well.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+
+Users can create various resources in {es} that they own, access, and manage.
+For example, using the <<async-search,async search API>> to create a search
+request, <<security-api-create-api-key,creating an API key>>, or activating a
+<<user-profile,user profile>>.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:15]: Strictly, users don't access or manage their own profiles (nor can they "create" them)

Perhaps we should say
```suggestion
Some types of resources in {es} are owned by a single user.
For example <<async-search,async search contexts>>,
<<security-api-create-api-key,API keys>>,
and <<user-profile,user profiles>> each have an owning user.
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+
+When a user creates a resource, {es} captures the user's `username` and realm
+information as part of the resource's metadata.
+When the user attempts to access the resource at a later time, {es} compares
+captured `username` and realm information against those from the accessing
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:20]:
```suggestion
the captured `username` and realm information against those from the accessing
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+user. The access is allowed only when both of them match.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:21]:
```suggestion
user. The access is not allowed unless both the realm and `username` match.
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+That is, if {es} detects that a `username` from two different realms is
+attempting to access a resource, {es} assumes that these users are distinct and
+doesn't allow resources to be shared between them.
+
+However, there are cases where the same end-user can authenticate with
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:26]:
```suggestion
However, there are cases where the same individual is able to authenticate with
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+be backed by the same directory service. Additionally,
+<<configuring-authorization-delegation,authorization delegation>> allows one
+realm to delegate authorization to another realm. If both realms authenticate
+end-users with the same `username`, it is reasonable to treat these users to
+be the same user.
+
+With security domains, you can configure multiple realms to be under the same
+domain, users can authenticate with any of the realms and have access to the
+same set of resources regardless of which realm under the same domain they're
+authenticating from. For example, a single <<user-profile,user profile>> is
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:38]: I don't feel strongly about this one, but the sentence was long and I suspect it could hard to follow for people with a lower proficiency in English.

```suggestion
same set of resources regardless of which realm authenticating from, 
(provided those realms are under the same domain).
For example, a single <<user-profile,user profile>> is
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+associated to the user which enables user preferences, notifications, and other
+user data to be shared across domains. The same user can view results from an
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:40]: I found the "user which enables" to be hard to follow.
We could change it to "user, enabling ..."

Or something more like this:
```suggestion
authenticating from.
For example, two users who are linked by a single domain will
share a <<user-profile,user profile>>, and can share user
preferences, notifications, and other user data. 
The same user can view results from an
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 23]
```diff
@@ -0,0 +1,97 @@
...
+asynchronous search request or a scrolling search. If the user has the
+necessary privileges, they can also view and manage API keys across realms.
+
+==== Configure a security domain
+
+You specify the realms for a domain in `elasticsearch.yml`. The realms must
+be defined in `elasticsearch.yml`, it doesn't have to be enabled to be listed
+under a domain.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:48]:
```suggestion
be defined in `elasticsearch.yml` (but do not need to be enabled).
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 24]
```diff
@@ -0,0 +1,97 @@
...
+. Restart {es}.
+
+[NOTE]
+====
+Though it's possible to configure different <<roles,roles>> for the same user when
+authenticating with different realms, it is _not_ recommended.
+Using <<configuring-authorization-delegation,authorization delegation>>
+is a good way to ensure that a user is assigned the same roles from multiple realms.
+
+{es} always enforces the user privileges for the authentication it has at the moment.
```
Comment (tvernum) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:85]:
```suggestion
{es} always enforces the privileges associated with the currently authenticated user.
```

File: x-pack/docs/en/security/authentication/user-profile.asciidoc (hunk 13 above)
Comment (tvernum) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc:32]: I'm aware that I'm correcting my own text here :)

File: x-pack/docs/en/security/authentication/security-domain.asciidoc (hunk 17 above)
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc:15]: Good catch!

File: x-pack/docs/en/security/authentication/security-domain.asciidoc (hunk 15 above)
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc:8]: I modified your version a little bit to say
> Security domains are a method of grouping different <<realms,realms>>
so that the {stack} can recognize when a single user authenticates with
these realms.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc (hunk 22 above)
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc]: I updated to `user, enabling ...` because your other version uses the term `link` and I am concerned this might be another cogonitive load for readers.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+doesn't allow resources to be shared between them.
+
+However, there are cases where the same user can authenticate with
+different realms and need to share the same set of resources across the realms.
+For example, an <<ldap-realm,LDAP realm>> and a <<saml-realm,SAML realm>> can
+be backed by the same directory service. Additionally,
+<<configuring-authorization-delegation,authorization delegation>> allows one
+realm to delegate authorization to another realm. If both realms authenticate
+users with the same `username`, it is reasonable to treat these users to
+be the same user.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:34]:
```suggestion
users with the same `username`, it is reasonable to treat these users as the
same end-user identity from a resource ownership point of view.
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+
+With security domains, you can configure multiple realms to be under the same
+domain, users can authenticate with any of the realms and have access to the
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:37]:
```suggestion
With security domains, you group multiple realms to be part of a domain. Users can authenticate with any of the domain realms and have access to the
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 27]
```diff
@@ -0,0 +1,98 @@
...
+same set of resources regardless of which realm authenticating from,
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:38]:
```suggestion
same set of resources regardless of which realm they authenticated with,
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+(provided those realms are under the same domain).
+For example, a single <<user-profile,user profile>> is
+associated to the user, enabling user preferences, notifications, and other
+user data to be shared across domains. The same user can view results from an
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:42]:
```suggestion
associated to the end-user identity, enabling preferences, notifications, and other
user data to be shared across domain realms. The single user can view results from an
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 29]
```diff
@@ -0,0 +1,98 @@
...
+asynchronous search request or a scrolling search. If the user has the
+necessary privileges, they can also view and manage API keys across realms.
+
+==== Configure a security domain
+
+You specify the realms for a domain in `elasticsearch.yml`. The realms must
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:48]:
```suggestion
You associate realms under a domain in `elasticsearch.yml`. All the domain realms must
```

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 30]
```diff
@@ -0,0 +1,98 @@
...
+be defined in `elasticsearch.yml` (but do not need to be enabled).
+Specifying either an undefined realm, synthetic realm, or the reserved realm
+under a domain will prevent the node from starting.
+A realm doesn't need to belong to a domain. If it does, the realm can only
+belong to a single domain. Specifying a realm in multiple domains generates an
+error when {es} starts.
```
Comment (albertzaharovits) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:54]:
```suggestion
Realms can be associated to at most one domain. It is an error ({es} fails to start) to assign an undefined realm to a domain or to assign a single realm to multiple domains.
```

**Review by albertzaharovits — APPROVED** [review]
LGTM
This is looking glorious, well done Yang and Tim!
I had a couple of suggestions, which are probably particular to my expression style. Take it or leave it, as you wish. 
I wish I had this sort of description before we went on coding it 😃

File: x-pack/docs/en/security/authentication/security-domain.asciidoc (hunk 26 above)
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc:37]: I prefer keep saying just `any of the realms` instead of `any of the domain realms` since the meaning is clear enough given the context. Even though it is clear to us what `domain realms` means, it could still be confusing to other readers. So I'd avoid it unless we need to formally introduce the term.
I took other changes. Thanks!

File: x-pack/docs/en/security/authentication/security-domain.asciidoc (hunk 30 above)
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc:54]: I reordered this section and moved it under the configuration example which has a second step says "Restart ES". It feels more naturally to talk about what could cause failure to start right after the instruction of restart. It also reads better to give example first before talking about error conditions.

**Comment by tvernum** [comment]
I'm happy to do another round if you need my input, but you're probably going to get more value from having @lockewritesdocs's editorial eye run over it.

Since @albertzaharovits has approved, I'm happy to just leave the final review to @lockewritesdocs, but ping me if you want me to look again.

**Comment by ywangd** [comment]
Thanks Tim. Let's have Adam to run a final round of this PR.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 31]
```diff
@@ -0,0 +1,99 @@
...
+Security domains don't bypass <<authorization,user authorization>> when
+resource sharing requires them. For example, a user requires the
+`manage_own_api_key` privilege to manage their own API keys. If that user does
+not have this privilege when authenticating with one realm, they won't be able
+to manage API keys while authenticating with this realm.
+This behaviour might be unexpected since the same user may or may not be able
+to perform the same operation depending on the realm it authenticates with.
+But it is a safe behaviour that ensures actual actions always comply to
+authorization requirements.
+====
```
Comment (lockewritesdocs) [inline x-pack/docs/en/security/authentication/security-domain.asciidoc:97]:
```suggestion
.Managing privileges across realms
***
Though it's possible to configure different <<roles,roles>> for the same user when
authenticating with different realms, it is _not_ recommended.
Using <<configuring-authorization-delegation,authorization delegation>>
is a good way to ensure that a user is assigned the same roles from multiple realms.

{es} always enforces the privileges associated with the currently authenticated user.
Security domains don't bypass <<authorization,user authorization>> when
resource sharing requires them. For example, a user requires the
`manage_own_api_key` privilege to manage their own API keys. If that user doesn't
have this privilege when authenticating with one realm, they won't be able
to manage API keys while authenticating with this realm.
This behavior might be unexpected since a single user may or may not be able
to perform the same operation depending on the realm it authenticates with.
However, it's a safe behavior that ensures actual actions always comply to
authorization requirements.
****
```

**Review by lockewritesdocs — COMMENTED** [review]
Left a few suggested changes. I think that the big change would be moving some of the overview content related to user profiles into the introductory section and focusing the "How user profiles work" content on the Kibana interaction.

File: x-pack/docs/en/security/authentication/security-domain.asciidoc [hunk 32]
```diff
@@ -0,0 +1,99 @@
...
+For example, a single <<user-profile,user profile>> is
+associated to the end-user, enabling preferences, notifications, and other
+user data to be shared across realms. The user can view results from an
+asynchronous search request or a scrolling search across realms. If the user has the
+necessary privileges, they can also view and manage API keys across realms.
+
+==== Configure a security domain
+
+You configure realms to be under a domain in `elasticsearch.yml`. The realms must
+be defined in `elasticsearch.yml` (but do not need to be enabled).
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/security-domain.asciidoc:49]: When a realm is defined in the yaml file, it is automatically enabled, but you can opt-in to disable it with `enabled: false`, e.g.:
```
ldap.ldap1:
  enabled: false
  ...
```

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 33]
```diff
@@ -0,0 +1,99 @@
+[role="xpack"]
+[[user-profile]]
+=== User profiles
+
+beta::[]
+
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc:6]: We briefly touched on this in a separate channel and agreed that this is the change we want. I have incorporated it in the latest commit. Thanks Adam!

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 34]
```diff
@@ -0,0 +1,99 @@
...
+Persistent means that a profile exists without the need for the user to be online.
+Stable means it has an unique identifier that does not change through
+the lifetime of a deployment and can be used to refer to the associated
+user.
+Being able to uniquely refer to users regardless whether they are actively
+online is a critical function that underpins important features like
+personalisation and collaboration in {kib}.
+
+==== How a user profile works
+
```
[the remark this answers is not in the document]
Comment (ywangd) [reply x-pack/docs/en/security/authentication/user-profile.asciidoc]: Same as above

**Comment by lockewritesdocs** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
CI failure is unrelated and I raised https://github.com/elastic/elasticsearch/issues/85887

**Comment by ywangd** [comment]
@elasticmachine update branch

File: x-pack/docs/en/security/authentication/user-profile.asciidoc [hunk 35]
```diff
@@ -0,0 +1,93 @@
...
+`roles`, and `realms`, and also includes the profile unique ID and timestamp of
+the operation. You can retrieve a user profile with
+the <<security-api-get-user-profile,get user profile API>> by including the
+profile's unique ID (`uid`).
+
+In addition to the user's basic information, you can add data to a profile document
+with the <<security-api-update-user-profile-data,update user profile API>>. For
+example, you can add user-specific preferences as part of the profile data.
+
+The <<security-api-suggest-user-profile,search user profile API>> can be used to search profiles
```
Comment (lockewritesdocs) [inline x-pack/docs/en/security/authentication/user-profile.asciidoc:72]:
```suggestion
Use the <<security-api-suggest-user-profile,search user profile API>> to search profiles
```

**Review by lockewritesdocs — APPROVED** [review]
Left a singular, minor comment that you can choose to incorporate. Thanks for the collaboration @ywangd!

**Comment by ywangd** [comment]
Failure is unrelated, I raised https://github.com/elastic/elasticsearch/issues/85903

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-2
